### PR TITLE
fix: pulling error when image has duplicated layers

### PR DESCRIPTION
**Pull request by Xynnn007**
When an image to be pulled has duplicated layers, the 2nd layer will write the same layer data and raise layer. This is because the time to update the database that stores layer metadata is after pulling all layers of an image. This commit
will try to construct a new layer list to be pulled via deduplicating the layers of the OciManifest.

Close https://github.com/confidential-containers/image-rs/issues/76

Signed-off-by: Xynnn007 <xynnn@linux.alibaba.com>

### CODE REVIEW

File: src/image.rs [hunk 1]
```diff
@@ -282,13 +282,19 @@ impl ImageClient {
             ));
         }
 
+        let mut layers = Vec::new();
```
Comment (arronwy) [inline src/image.rs:285]: Maybe use `unique_layers` will be more straightforward

**Review by arronwy — APPROVED** [review]
Thanks @Xynnn007 LGTM!

File: src/image.rs (hunk 1 above)
Comment (Xynnn007) [reply src/image.rs]: Thanks, much clearer! fixed it

**Comment by Xynnn007** [comment]
The nightly error occurs because of crate `rstest`, and it is being handled by the crate's owner now. I think we can fix it some days later in another pr.

**Comment by arronwy** [comment]
> The nightly error occurs because of crate `rstest`, and it is being handled by the crate's owner now. I think we can fix it some days later in another pr.

Yes, nightly CI error is not related to this PR: https://github.com/la10736/rstest/issues/169

**Review by jialez0 — APPROVED** [review]

**Pull request merged**